### PR TITLE
Catch the proper exceptions in solfuzzer

### LINKED ISSUE #4458: Fuzzer catches too many exceptions
The fuzzer currently does not fail on failing assertions.

Example code (from #4417):
```
pragma solidity ^0.4.24;

contract SomeContract {

  uint public balance = 0;


  function balance(uint number) public {

  }

  function doSomething() public {
    balance(3);
  }
}
```

-- found by @mattaereal

### PR DESCRIPTION
**Pull request by axic**
Fixes #4458.

### CODE REVIEW

File: test/tools/fuzzer.cpp [hunk 1]
```diff
@@ -135,6 +135,10 @@ void testCompiler(bool optimize)
 		for (Json::Value const& error: outputJson["errors"])
 		{
 			string invalid = contains(error.asString(), vector<string>{
+				// StandardJSON error types
+				"Exception",
+				"InternalCompilerError",
+				// Old-school error messages
 				"Internal compiler error",
```
Comment (axic) [inline test/tools/fuzzer.cpp:142]: All of these could be removed below.

**Review by chriseth — DISMISSED** [review]

**Comment by axic** [comment]
This discovered a failure in the docs:
```
		pragma experimental ABIEncoderV2;
		contract Test {
			enum ActionChoices { GoLeft, GoRight, GoStraight, Sit }
			struct Simple { uint i; }
			struct Nested { X[2][] a; uint y; }
			struct X { bytes32 x; Test t; Simple[] s; }
			function f(ActionChoices, uint, Simple) external {}
			function g(Test, Nested) external {}
			function h(function(Nested) external returns (uint)[]) external {}
			function i(Nested[]) external {}
		}
```

```
Internal compiler error during compilation:
/Users/alex/Projects/solidity/libsolidity/codegen/ABIFunctions.cpp(1191): Throw in function std::__1::string dev::solidity::ABIFunctions::abiDecodingFunctionCalldataArray(const dev::solidity::ArrayType &)
Dynamic exception type: boost::exception_detail::clone_impl<dev::solidity::InternalCompilerError>
std::exception::what: 
[dev::tag_comment*] = 
```

**Comment by chriseth** [comment]
Calldata arrays with non-value base types are not yet supported by Solidity. I will change it to a solUnimplemented exception.

**Comment by chriseth** [comment]
Rebased on top of the fix to check if it does it.

**Review by chriseth — APPROVED** [review]

**Pull request merged**